File: 2024/javascript/src/parsing.js
/**
 * @param {string} data
 * @returns {string[]}
 */
function asLines(data) {
  return data
    .trim()
    .split("\n")
    .map((line) => line.trim());
}

/**
 * @param {string} input_data
 * @returns {number[][]}
 */
function groupAsInts(input_data) {
  const lines = asLines(input_data);
  return lines.map((line) => line.split(/\s+/).map((num) => Number(num)));
}

module.exports = { groupAsInts };

File: 2024/python/src/parsing.py
from typing import Iterator


# A running collection of functions to help parse problem input. Will be
# interesting to see how this looks later on and how well I'm able to reuse them.


def as_lines(data: str) -> Iterator[str]:
    return map(str.strip, data.strip().splitlines())


# Doing this with iterators was getting more difficult to work with. Going to
# just use lists for now. Maybe will use the iterator version again
def group_as_ints(input_data: str) -> list[list[int]]:
    lines = as_lines(input_data)
    return [[int(num) for num in line.split()] for line in lines]


def group_as_ints_iter(input_data: str) -> Iterator[Iterator[int]]:
    lines = as_lines(input_data)
    return ((int(num) for num in line.split()) for line in lines)
